File: leet/esji/1768.Merge_Strings_Alternately.go
package main

import (
	"fmt"
)

func mergeAlternately(word1 string, word2 string) string {
	var answer string = ""
	var index int = 0

	for {
		if len(word1) == index {
			answer += string(word2[index:])
			break
		} else if len(word2) == index {
			answer += string(word1[index:])
			break
		} else {
			fmt.Println(word1[index] + word2[index])
			answer += string(word1[index]) + string(word2[index])
		}
		index += 1
	}
	return answer
}

func main() {
	fmt.Println(mergeAlternately("ab", "qwe"))
}

File: leet/esji/1768.Merge_Strings_Alternately.py
from typing import List


class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result_str = ""
        i : int = 0

        while(True):
            if len(word1) == i:
                result_str += word2[i:]
                break
            elif len(word2) == i:
                result_str += word1[i:]
                break
            else:
                result_str += word1[i] + word2[i]
            i += 1

        return result_str


if __name__ == "__main__":
    s = Solution()
    print(s.mergeAlternately("ab", "qwe"))